src/stores/asynchronousAction.ts

```diff
@@ -10,13 +14,21 @@ export interface IAsynchronousAction<S, R> {
   // and it is not a real "action" that a reducer will ever see.
   type: string;
   types: IActionTypeCollection;
-  doPerform: (state: S) => Promise<R> | undefined;
+  doPerform: (
+    state: S,
+    dispatch: ThunkDispatch<IState, void, AnyAction>
+  ) => Promise<R> | undefined;
 }
 
 export interface IAsynchronousActionParams<P, S, R> {
   actionTypePrefix: string;
-  perform: (state: S, payload?: P) => Promise<R>;
+  perform: (
+    state: S,
+    dispatch: ThunkDispatch<IState, void, AnyAction>,
+    payload?: P
+  ) => Promise<R>;
   shouldPerform: (state: S) => boolean;
+  throwOnError: boolean;
```

Comment:
While creating our action we can now define if it should throw or not.

I guess the goal will be to not need throwing, but for now to avoid to many steps in the refactoring at once, I added this as an option. That way when we are done moving everything it will also be easy to identify which actions and bits of the UI need further refactoring.